### ISSUE TITLE
Can't get to work with Pleroma/Rebased/Akkoma, claims token exists on first try which fails

### ISSUE THREAD

**Issue by guysoft** [halcy/Mastodon.py#331]
Tested and got same results on akkoma and rebased.

On a mastodon instance it works, fine.
If there any way to get more debug output what the requests are would be glad to provide.

Reproduction code:

```python
from mastodon import Mastodon
import sys
scopes = ('read',)
user = "guysoft@hayu.sh"
username, domain = user.split("@")
client_secret = domain + ".client.secret"
url = "https://" + domain
user_secret = domain + ".user." + username + ".secret"
        

Mastodon.create_app("fedigroup1",
    api_base_url=url,
    scopes=scopes,
    to_file=client_secret
)
    
mastodon = Mastodon(client_id=client_secret, api_base_url=url)
url = mastodon.auth_request_url(client_id=client_secret, scopes=scopes)


print("Visit the following URL and authorize the app:")
print(url)

print("Then paste the access token here:")
token = sys.stdin.readline().rstrip()
client = mastodon.log_in(code=token, to_file=user_secret, scopes=scopes)
print(mastodon)
print(dir(mastodon))
```

Output:

```python
python3 ./mastodon_login_minimal.py 
Visit the following URL and authorize the app:
https://akkoma.gnethomelinux.com/oauth/authorize?client_id=XXX&response_type=code&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=read&force_login=False&state=None&lang=None
Then paste the access token here:
O_XXX
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/mastodon/authentication.py", line 322, in log_in
    response = self.__api_request('POST', '/oauth/token', params, do_ratelimiting=False)
  File "/usr/local/lib/python3.10/dist-packages/mastodon/internals.py", line 299, in __api_request
    raise ex_type('Mastodon API returned error', response_object.status_code, response_object.reason, error_msg)
mastodon.errors.MastodonAPIError: ('Mastodon API returned error', 400, 'Bad Request', 'Invalid credentials')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./mastodon_login_minimal.py", line 26, in <module>
    client = mastodon.log_in(code=token, to_file=user_secret, scopes=scopes)
  File "/usr/local/lib/python3.10/dist-packages/mastodon/authentication.py", line 330, in log_in
    raise MastodonIllegalArgumentError('Invalid access token or redirect_uris: %s' % e)
mastodon.errors.MastodonIllegalArgumentError: Invalid access token or redirect_uris: ('Mastodon API returned error', 400, 'Bad Request', 'Invalid credentials')
```

Notes:
Both akkoma and rebased don't actually ask you to approve the request, they just post:
Authorization exists

With a token.

If you need a rebased/akkoma instance to test contact me and I can give you access to mine :)

Version 1.8.0

**Comment by guysoft** [comment]
Ok, thanks for the ``debug_requests=True`` pointer, that helped me by letting me view the requests and eventually figuring out how to get it to work.

TL;DR
The line
```python
url = mastodon.auth_request_url(client_id=client_secret, scopes=scopes)
```
Needs to be replaced with:
```python
url = mastodon.auth_request_url(client_id=client_secret, scopes=scopes, force_login=True)
```

It looks like pleroma instances will not work with ```force_login=False``` which is the default.

Also for future reference here is a shell script to get a token using curl which helped me to compare:
https://gist.github.com/a1batross/11f3c3026aacb580305898d0c191e7de

Full code:
```python3
from mastodon import Mastodon
import sys
# scopes = ('read',)
scopes = ("read",)
user = "guysoft@hayu.sh"
username, domain = user.split("@")
client_secret = domain + ".client.secret"
url = "https://" + domain
user_secret = domain + ".user." + username + ".secret"
        

Mastodon.create_app("fedigroup1",
    api_base_url=url,
    scopes=scopes,
    to_file=client_secret
)
    
mastodon = Mastodon(client_id=client_secret, api_base_url=url, debug_requests=True)
url = mastodon.auth_request_url(client_id=client_secret, scopes=scopes, force_login=True)


print("Visit the following URL and authorize the app:")
print(url)

print("Then paste the access token here:")
token = sys.stdin.readline().rstrip()
client = mastodon.log_in(code=token, to_file=user_secret, scopes=scopes)
print(mastodon)
print(dir(mastodon))
```
